src/uu/pgrep/src/pgrep.rs

```diff
@@ -0,0 +1,278 @@
+// This file is part of the uutils procps package.
+//
+// For the full copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+
+pub mod pid;
+
+use clap::{arg, crate_version, Arg, ArgAction, ArgGroup, ArgMatches, Command};
+use pid::{walk_pid, PidEntry, TerminalType};
+use std::{borrow::BorrowMut, cmp::Ordering, collections::HashSet};
+use uucore::{error::UResult, format_usage, help_about, help_usage};
+
+const ABOUT: &str = help_about!("pgrep.md");
+const USAGE: &str = help_usage!("pgrep.md");
+
+#[uucore::main]
+pub fn uumain(args: impl uucore::Args) -> UResult<()> {
+    let matches = uu_app().try_get_matches_from(args)?;
+    let patterns = collect_arg_patterns(&matches);
+
+    let Some(result) =
+        collect_oldest_newest(&matches, &patterns).or(collect_normal_pid(&matches, &patterns))
+    else {
+        println!("pgrep: no matching criteria specified");
+        println!("Try `pgrep --help' for more information");
+        return Ok(());
```

Comment:
This is an error situation and hence should return an error. You can probably use an `USimpleError`.